__tests__/app.test.js

```diff
@@ -156,5 +156,66 @@ describe("news-api", () => {
                 })
             })
         })
+        describe("GET /api/articles/:article_id/comments", () => {
+            test("responds with status code 200 and an object in expected format", () => {
+                return request(app)
+                .get("/api/articles/1/comments")
+                .expect(200)
+                .then(({body}) => {
+                    expect(body).toHaveProperty("comments")
+                    const { comments } = body
+                    expect(comments instanceof Array).toBe(true)
+                    expect(comments.length).toBeGreaterThan(0)
+                    comments.forEach((element) => {
+                        expect(element instanceof Object).toBe(true)
+                    })
+                })
+            })
+            test("each object in the 'comments' array has the expected keys and value types", () => {
+                return request(app)
+                .get("/api/articles/1/comments")
+                .expect(200)
+                .then(({body}) => {
+                    const { comments } = body
+                    expect(comments.length).toBeGreaterThan(0)
+                    comments.forEach((comment) => {
+                        expect(comment.comment_id).toEqual(expect.any(Number))
+                        expect(comment.votes).toEqual(expect.any(Number))
+                        expect(comment.created_at).toEqual(expect.any(String))
+                        expect(comment.author).toEqual(expect.any(String))
+                        expect(comment.body).toEqual(expect.any(String))
+                        expect(comment.article_id).toEqual(expect.any(Number))
+                    })
+                })
+            })
+            test("responds with status code 404 'not found' if there are no comments with a matching article_id", () => {
+                return request(app)
+                .get("/api/articles/9999/comments")
+                .expect(404)
+                .then(({body}) => {
+                    const { msg } = body
+                    expect(msg).toBe("not found")
+                })
+            })
+            test("responds with status code 400 'bad request' when provided an article_id that is not a number", () => {
+                return request(app)
+                .get("/api/articles/not-a-number/comments")
+                .expect(400)
+                .then(({body}) => {
+                    const { msg } = body
+                    expect(msg).toBe("bad request")
+                })
+            })
+            test("the 'articles' array is returned in descending date order", () => {
```

Comment:
This is a really nice test! Please do group it in with your other 'happy path' tests though.